backend/model/jobValidator.js

```diff
@@ -96,68 +92,94 @@ class JobValidator extends Job {
   // Helpers
   async __checkDisk(size, target) {
     /*
-		This does not account for size on disk(blocks used) vs file size, for
-		larger files this may be an issue.
+    This does not account for size on disk(blocks used) vs file size, for
+    larger files this may be an issue.
 
-		This also not not account for space needed to extract or decrypt
-		operations.
-		*/
+    This also not not account for space needed to extract or decrypt
+    operations.
+    */
 
     return (await checkDiskSpace(target)).free > size;
   }
 
   /*
-	Actions
+  Actions
 
-	The client can initiate actions against the contract as a validator. Most of
-	these result in a action being emitted to the smart contract. 
-	*/
+  The client can initiate actions against the contract as a validator. Most of
+  these result in a action being emitted to the smart contract.
+  */
 
   async downloadAndTestModel(event) {
+
     try {
       let job = event.returnValues;
 
       this.downloadPath = `${conf.appDownloadPath}${this.jobData.jobPoster}/${this.id}`;
 
+      console.log('download path:')
+      console.log(this.downloadPath);
+
       // Make sure download spot exists
       fs.ensureDirSync(this.downloadPath);
 
       // Download the shared files
-      await webtorrent().downloadAll(
+      let downloads = await webtorrent().downloadAll(
         this.downloadPath,
         event.returnValues.trainedModelMagnetLink,
         event.returnValues.testingDatasetMagnetLink,
         event.returnValues.untrainedModelMagnetLink
       );
 
-      //Test the modal and get loss
-      const std = await exec(
-        "python3 unsorted/validator_node.py 2> /dev/null | tail -n 1"
+      console.log("Downloads", downloads);
+
+      console.info(
+        "Download done!",
+        this.instanceId,
+        new Date().toLocaleString()
       );
 
-      console.log("Python STDOUT: ", std);
+      let jupyterNotebookPathname;
+      let testingDataPathname;
+      let pythonPathname;
+
+      for (let download of downloads) {
+        if (fileExtensionExtractor(download.dn) == "ipynb") {
+          jupyterNotebookPathname = download.path + "/" + download.dn;
```

Comment:
NIT: in the future, we can try to turn this into its own helper. So we don't have to concatenate it ourselves here